### ISSUE TITLE
[Feature] Reveal in Explorer

### ISSUE THREAD

**Issue by ShlomiRex** [jupyterlab/jupyterlab#9458]
### Problem

There are some files that I want to explore in File Explorer (open diffirent programs for it, edit it) that jupyterlab just can't do.

![image](https://user-images.githubusercontent.com/17744127/101907868-03bc9a80-3b70-11eb-8d8d-dd34e9789223.png)

### Proposed Solution

Add 'Reveal in Explorer' option when right click onto file, directory, or empty container in the 'File explorer' of jupyterlab.

### Additional context

For example, in VSCode:

![image](https://user-images.githubusercontent.com/17744127/101908026-47af9f80-3b70-11eb-8085-2563cd0c2604.png)

**Comment by jasongrout** [comment]
Thanks for the suggestion. A couple of thoughts:

1. For many people running JupyterLab, the server is on a different computer than the browser, so opening a file in the local computer's file browser does not make sense (the files are not there, they are on the remote server). So first, we'd have to be able to tell if the server is on the same computer. Supposing we could...
2. As far as I can tell, you cannot open a file in the computer's file browser directly from a web browser. I think that is usually considered a security risk if a browser could do that. One way to deal with this is to send a request back to the local server asking the *server* to open the file locally....
3. Even if the server is on the same computer, we'd want to make sure the server is running as the local user, and has access to desktop resources, so showing a file made sense. If the server was, say, on the local computer but running as a different user, we wouldn't want *their* file browser popping up. Furthermore, if the server was started from, say, the Linux command line, it might not know the right desktop to open a file browser in, etc.

Basically, the constraints of JupyterLab (running in the browser, connecting to a remote server) are really working against something like this.

This would be much more manageable in an electron app running JupyterLab (like VS Code). Then you know the server is local, you have direct access to the filesystem, etc. So I would say this is in scope for something like https://github.com/jupyterlab/jupyterlab_app, but probably would be prohibitive to do in JupyterLab itself, so I'll close this issue - feel free to open one in https://github.com/jupyterlab/jupyterlab_app.

That said, if someone can find a way to do it well in JupyterLab in a browser-independent way, I'm happy to discuss an implementation, and we can certainly reopen this issue if it seems there is a path forward.

**Comment by jasongrout** [comment]
One thing we can do is to get a path to the file on the local disk in some cases with certain assumptions. You could paste that into your local file browser. Would that help?
